python/fledge/plugins/filter/rename/rename.py

```diff
@@ -0,0 +1,180 @@
+        'displayName': 'Replace With'
+    },
+    'enable': {
+        'description': 'Enable/Disable filter plugin',
+        'type': 'boolean',
+        'default': 'false',
+        'displayName': 'Enabled',
+        'order': "5"
+    }
+}
+
+
+def plugin_info():
+    """ Returns information about the plugin
+    Args:
+    Returns:
+        dict: plugin information
+    Raises:
+    """
+    return {
+        'name': PLUGIN_NAME,
+        'version': '1.9.1',
+        'mode': "none",
+        'type': 'filter',
+        'interface': '1.0',
+        'config': _DEFAULT_CONFIG
+    }
+
+
+def plugin_init(config, ingest_ref, callback):
+    """ Initialise the plugin
+    Args:
+        config:     JSON configuration document for the Filter plugin configuration category
+        ingest_ref: filter ingest reference
+        callback:   filter callback
+    Returns:
+        data:       JSON object to be used in future calls to the plugin
+    Raises:
+    """
+    data = copy.deepcopy(config)
+    global the_callback, the_ingest_ref
+    the_callback = callback
+    the_ingest_ref = ingest_ref
+    return data
+
+
+def plugin_reconfigure(handle, new_config):
+    """ Reconfigures the plugin
+
+    Args:
+        handle:     handle returned by the plugin initialisation call
+        new_config: JSON object representing the new configuration category for the category
+    Returns:
+        new_handle: new handle to be used in the future calls
+    """
+    _LOGGER.info("Old config {} \n new config {} for {} plugin ".format(handle, new_config, PLUGIN_NAME))
+    new_handle = copy.deepcopy(new_config)
+    return new_handle
+
+
+def plugin_shutdown(handle):
+    """ Shutdowns the plugin doing required cleanup.
+
+    Args:
+        handle: handle returned by the plugin initialisation call
+    Returns:
+    """
+    global the_callback, the_ingest_ref
+    the_callback = None
+    the_ingest_ref = None
+    _LOGGER.info('{} filter plugin shutdown.'.format(PLUGIN_NAME))
+
+
+def plugin_ingest(handle, data):
+    """ Modify readings data and pass it onward
+
+    Args:
+        handle: handle returned by the plugin initialisation call
+        data:   readings data
+    """
+    global the_callback, the_ingest_ref
+    if handle['enable']['value'] == 'false':
+        # Filter not enabled, just pass data onwards
+        filter_ingest.filter_ingest_callback(the_callback, the_ingest_ref, data)
+        return
+
+    # Filter is enabled: compute for each reading
+    processed_data = {}
+    for element in data:
+        processed_data = find_and_replace(handle['operation']['value'], handle['find']['value'],
+                                          handle['replaceWith']['value'], element)
+    _LOGGER.debug("processed data {}".format([processed_data]))
+    # Pass data onwards
+    filter_ingest.filter_ingest_callback(the_callback, the_ingest_ref, [processed_data])
+
+    _LOGGER.info("{} filter ingest done".format(PLUGIN_NAME))
```

Comment:
better not called ingest (in logs at-least) what if applied on north. We should actually avoid this log statement as INFO (as of plugin) for performance reasons, can keep as debug/ better commented.